diff --git a/src/video/cocoa/SDL_cocoaopengl.m b/src/video/cocoa/SDL_cocoaopengl.m
--- a/src/video/cocoa/SDL_cocoaopengl.m
+++ b/src/video/cocoa/SDL_cocoaopengl.m
@@ -133,6 +133,12 @@
         }
     }
 
+    if(_this->gl_config.major_version >= 3)    
+    {    
+        attr[i++] = NSOpenGLPFAOpenGLProfile; 
+        attr[i++] = NSOpenGLProfileVersion3_2Core; 
+    }     
+    
     attr[i++] = NSOpenGLPFAScreenMask;
     attr[i++] = CGDisplayIDToOpenGLDisplayMask(displaydata->display);
     attr[i] = 0;
